### ISSUE TITLE
PSR12.Classes.AnonClassDeclaration incorrectly enforcing that closing brace be on a line by itself

### ISSUE THREAD

**Issue by dylan-kerr** [squizlabs/PHP_CodeSniffer#2678]
There appears to be no accepted way to format anonymous classes. This broke between 3.4.2 and 3.5.1.

I am using the following ruleset:
```xml
    <rule ref="PSR12"/>
    <rule ref="Generic.Arrays.DisallowLongArraySyntax"/>
    <rule ref="Generic.Formatting.SpaceAfterNot">
        <properties>
            <property name="spacing" value="0"/>
        </properties>
    </rule>
```

I would normally format anonymous classes like the following example:
```php
<?php

namespace Example;

class Example
{
    public function __construct()
    {
        $this->foo(new class {
            public function bar()
            {
            }
        }); // <-- New error message: "Closing class brace must be on a line by itself"
    }

    public function foo($input)
    {
    }
}
```

There doesn't seem to be any way to solve this while still satisfying other rules. If I move the closing parenthesis onto the next line, I get error message "Expected 0 spaces before closing bracket; newline found".

**Comment by dylan-kerr** [comment]
This might be related to issue #2621 or PR #2622.

I have also found a layout that phpcs does not complain about:
```php
<?php

namespace Example;

class Example
{
    public function __construct()
    {
        $this->foo(
            new class
            {
                public function bar()
                {
                }
            }
        );
    }

    public function foo($input)
    {
    }
}
```

I'm happy with this as a workaround, but others might consider this a breaking change.

**Comment by gsherwood** [comment]
> This broke between 3.4.2 and 3.5.1

The reason for this is because the included PSR12 standard was only about 50% complete until 3.5.0, so you may have also noticed a lot more errors being generated.

This issue is caused by the reuse of the PSR2 sniff to help aid in class checking. I think PSR-12 just needs to ignore this specific error code, but I'll make sure it is still enforcing the closing brace placement rules that the anon class formatting is inheriting for closures.

**Comment by gsherwood** [comment]
This is now fixed and both those code snippets pass without error.

I made the change in the PSR2 sniff directly as PSR2 doesn't mention anon classes so it probably shouldn't be that strict about them there. That had the effect of fixing PSR12 as well.

Thanks for reporting this.
